crates/ark-contracts/arkchain/src/orderbook.cairo

```diff
@@ -291,19 +295,77 @@ mod orderbook {
             };
         }
 
-        fn fullfil_order(
-            ref self: ContractState,
-            order_hash: felt252,
-            execution_info: ExecutionInfo,
-            signer: Signer
-        ) {}
+        fn fulfill_order(ref self: ContractState, execution_info: ExecutionInfo, signer: Signer) {
+            let order_hash = execution_info.order_hash;
+            let execution_hash = execution_info.hash();
+            SignerValidator::verify(execution_hash, signer);
+
+            // Check if order exists
+            let order: OrderV1 = match order_read(order_hash) {
+                Option::Some(o) => o,
+                Option::None => panic_with_felt252(orderbook_errors::ORDER_NOT_FOUND),
+            };
+
+            // Check if order offerer is not execution_info fulfiller
+            assert(order.offerer != execution_info.fulfiller, orderbook_errors::ORDER_SAME_OFFERER);
```

Comment:
Not sure this is a good test. In the case of Auction, the `offerer` and the `fulfiller` **MUST** be the same. As we will also verify the signatures.

Reply:
Wdym by 
> As we will also verify the signatures.

i think it's still important to check for difference or equality, should we do it in solis ?

Reply:
i moved it in the fulfull listing function and maybe we can add a test for the auction to check specifically that the offerer is the fulfiller